cmds/time_sos/service.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"os/exec"
+	"strings"
+)
+
+type TimeService struct {
+	Date string
+	Time string
+}
+
+func getCurrentDate() string {
+	tempDate, err := exec.Command("date", "+%F").Output()
+	if err != nil {
+		return ""
+	}
+	date := strings.TrimSpace(string(tempDate[:]))
+	// date must be in YYYY-MM-DD format, but this command can return single-digit months.
+	// If applicable, add a zero.
+	if len(date) < 10 {
+		date = date[:5] + "0" + date[5:]
+	}
+	return date
+}
+
+func getCurrentTime() string {
+	tempTime, err := exec.Command("date", "+%H:%M").Output()
```

Comment:
Get the time directly, the time package makes it easy. Every external command you call is a chance for failure.